clang/test/SemaSYCL/sycl-kernel-entry-point-attr-grammar.cpp

```diff
@@ -0,0 +1,137 @@
+// RUN: %clang_cc1 -triple x86_64-linux-gnu -std=c++17 -fsyntax-only -fsycl-is-device -verify %s
```

Comment:
I don't think I see tests for all of the things in your documented list.

Reply:
Those tests are included with the diagnostic work that will be in the next PR. See [here](https://github.com/tahonermann/llvm-project/compare/sycl-upstream-fe-pr1...tahonermann:llvm-project:sycl-upstream-fe-pr2) for what has been completed. We still have some SFINAE issues to address (an invalid use of the attribute is triggering SFINAE and therefore affecting overload resolution) and I'm still working on a diagnostic for forward declarability of kernel names (for SYCL portability warnings; Clang won't actually require forward declarability since it won't depend on integration headers).

Reply:
Erich, I'm going to resolve this conversation regarding tests and diagnostics (planned for the next PR). Please unresolve if there is more you want to see in this PR.

Reply:
I think that is fine for me.  Again, sorry for not seeing this one :/  I would LOVE to see the list of 'things not diagnosed' documented/updated somewhere (so that we can check to make sure we get them all, and more importantly (hah!) I can do so while reviewing).  But I'm OK holding off on this.